Tests/Common/Securities/Futures/FuturesExpiryFunctionsTests.cs

```diff
@@ -16,12 +16,189 @@
+            }
+        }
+
+        /// <summary>
+        /// Utility method to do the common work for all the symbols i.e. asserting the 
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <param name="dayTime"></param>
+        /// <param name="market"></param>
+        [TestCase(QuantConnect.Securities.Futures.Grains.Wheat, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Grains.Corn, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Grains.Soybeans, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Grains.SoybeanMeal, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Grains.SoybeanOil, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Grains.Oats, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.GBP, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.CAD, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.JPY, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.CHF, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.EUR, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.AUD, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Currencies.NZD, nineSixteenCentralTime)]
+        [TestCase(QuantConnect.Securities.Futures.Energies.CrudeOilWTI, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Energies.HeatingOil, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Energies.Gasoline, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Energies.NaturalGas, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Financials.Y30TreasuryBond, twelveOne)]
+        [TestCase(QuantConnect.Securities.Futures.Financials.Y10TreasuryNote , twelveOne)]
+        [TestCase(QuantConnect.Securities.Futures.Financials.Y5TreasuryNote, twelveOne)]
+        [TestCase(QuantConnect.Securities.Futures.Financials.Y2TreasuryNote, twelveOne)]
+        [TestCase(QuantConnect.Securities.Futures.Indices.SP500EMini,nineThirtyEasternTime)]
+        [TestCase(QuantConnect.Securities.Futures.Indices.NASDAQ100EMini, nineThirtyEasternTime)]
+        [TestCase(QuantConnect.Securities.Futures.Indices.Dow30EMini, nineThirtyEasternTime)]
+        [TestCase(QuantConnect.Securities.Futures.Meats.LiveCattle, twelveOclock)]
+        [TestCase(QuantConnect.Securities.Futures.Meats.LeanHogs, twelveOclock)]
+        [TestCase(QuantConnect.Securities.Futures.Meats.FeederCattle, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Metals.Gold, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Metals.Silver, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Metals.Platinum, zero)]
+        [TestCase(QuantConnect.Securities.Futures.Metals.Palladium, zero)]
+        public void TestExpiryDateFunction(string symbol, string dayTime)
```

Comment:
It could be a good idea to split the test cases up in an effort to easily diagnose what is going wrong if one of the test cases fails.  So maybe, a `TestMetalsExpiryDates`, `TestMeatsExpiryDates`, etc.